src/pages/signin/SignInModal.js

```diff
@@ -24,7 +24,7 @@ function SignInModal() {
             shouldEnableMaxHeight
             testID={SignInModal.displayName}
         >
-            <HeaderWithBackButton />
+            <HeaderWithBackButton onBackButtonPress={Navigation.dismissModal} />
```

Comment:
This caused inconsistency between browser back button and modal back button. (Coming from https://github.com/Expensify/App/issues/33417)